### PR TITLE
CICD worklow Travis

### PR DESCRIPTION
**Pull request by Surfict**
This PR add the following features : 

1. Travis build, tests and release stages

2. Before the release, the docker image is tagged with the followings elements, for the version 1.3.5 : 
   - 1
   - 1.3
   - 1.3.5
   - Commit hash
   - lastest

3. The version is checked before the release stage. If the version is <= to the DockerHub versionm the release stage fail.

4. Travis will build only for the projects that has been modified. ( Well not really, but it will stop on the before_install script)

5. The folder .github has been added and the different owner of the differents projects too.

6. Defined a common layout for every projects in example-config

7. Push to DockerHub can be done only from the master branch of the ITIS reposistory.

Related to #20

### CODE REVIEW

File: devpi/Makefile [hunk 1]
```diff
@@ -0,0 +1,92 @@
+PREDEFINED_VARIABLES := $(.VARIABLES)
+
+DOCKER_REPO        = ITISFoundation
+APP_NAME           = $(notdir $(PWD))
+DOCKER_IMAGE_NAME  = $(DOCKER_REPO)/$(APP_NAME)
+DOCKER_IMAGE_VERSION_FULL = $(shell cat VERSION)
+DOCKER_IMAGE_VERSION_TWO_DIGITS = $(shell cat VERSION | grep -Eo '^([0-9]+).([0-9])+')
```
Comment (oetiker) [inline devpi/Makefile]:
```suggestion
DOCKER_IMAGE_VERSION_TWO_DIGITS = $(shell cat VERSION | grep -Eo '^([0-9]+)\.([0-9])+')
```

File: check_versions.sh [hunk 2]
```diff
@@ -0,0 +1,39 @@
+#!/bin/bash
+set -e
```
Comment (sanderegg) [inline check_versions.sh]:
```suggestion
#!/bin/bash
# http://redsymbol.net/articles/unofficial-bash-strict-mode/
set -euo pipefail
IFS=$'\n\t'

```

File: ci/travis/helpers/check_changes.sh [hunk 3]
```diff
@@ -0,0 +1,21 @@
+#
```
Comment (sanderegg) [inline ci/travis/helpers/check_changes.sh]:
```suggestion
#!/bin/bash
# http://redsymbol.net/articles/unofficial-bash-strict-mode/
set -euo pipefail
IFS=$'\n\t'

```

File: .travis.yml [hunk 4]
```diff
@@ -37,3 +58,42 @@ jobs:
...
+        - unbuffer bash ci/travis/tests/devpi script
+      after_success:
+        - unbuffer bash ci/travis/tests/devpi after_success
+
+    - stage: release
+      if: NOT type = pull_request and branch = master
+      name: devpi release
+      language: python
+      sudo: required
+      python:
```
Comment (pcrespov) [inline .travis.yml]: Why would you need python? Is not only docker required?

Comment (sanderegg) [reply .travis.yml]: I think one can use generic for language. but if you don't set any then you'll get ruby which also looks weird ;)

File: devpi/.TAG [hunk 5]
```diff
@@ -0,0 +1 @@
+0.0.20191113-144306
```
[the remark this answers is not in the document]
Comment (Surfict) [reply devpi/.TAG]: This has been generated during tests and added by error to the repo, I will delete it

File: .travis.yml (hunk 4 above)
Comment (Surfict) [reply .travis.yml]: I changed for miminal, who doesn't have any language in particular (https://docs.travis-ci.com/user/languages/minimal-and-generic/)

File: example-config/Dockerfile [hunk 6]
```diff
@@ -0,0 +1 @@
+TODO
```
[the remark this answers is not in the document]
Comment (Surfict) [reply example-config/Dockerfile]: Maybe i could create a script what would copy  past the folder and fill the TODOs, except for the Makefile ?

File: devpi/docker-compose.yml [hunk 7]
```diff
@@ -1,9 +1,11 @@
...
   devpi:
-    image: itisfoundation/devpi:latest
+    image: ${DOCKER_REPO}/devpi:${VERSION:-latest}
     build:
       context: .
+      cache_from:
+        - itisfoundation/devpi:latest
       labels:
         org.label-schema.schema-version: "1.0"
         org.label-schema.build-date: "${BUILD_DATE}"
```
[the remark this answers is not in the document]
Comment (Surfict) [reply devpi/docker-compose.yml]: Done

**Pull request merged**